docs/codes-other-values/hold-reason-codes.rst

```diff
@@ -0,0 +1,247 @@
+    |                                  | Spot instance request               | 19                       |
+    |                                  | purged before instance              |                          |
+    |                                  | ID acquired.                        |                          |
+    +----------------------------------+-------------------------------------+--------------------------+
+    | | 43                             | Pre script failed.                  |                          |
+    | | [PreScriptFailed]              |                                     |                          |
+    +----------------------------------+-------------------------------------+--------------------------+
+    +----------------------------------+-------------------------------------+--------------------------+
+    | | 44                             | Post script failed.                 |                          |
+    | | [PostScriptFailed]             |                                     |                          |
+    +----------------------------------+-------------------------------------+--------------------------+
+    | | 45                             | Test of singularity runtime failed  |                          |
+    | | [SingularityTestFailed]        | before launching a job              |                          |
+    +----------------------------------+-------------------------------------+--------------------------+
+    | | 46                             | The job's allowed duration was      |                          |
+    | | [JobDurationExceeded]          | exceeded.                           |                          |
+    +----------------------------------+-------------------------------------+--------------------------+
+    | | 47                             | The job's allowed execution time    |                          |
+    | | [JobExecuteExceeded]           | was exceeded.                       |                          |
+    +----------------------------------+-------------------------------------+--------------------------+
+    | | 48                             | Prepare job shadow hook failed      |                          |
+    | | [HookShadowPrepareJobFailure]  | when it was executed;               |                          |
+    |                                  | status code indicated job should be |                          |
+    |                                  | held.                               |                          |
+    +----------------------------------+-------------------------------------+--------------------------+
+
+    Note for hold codes 12 [TransferOutputError] and 13 [TransferInputError]:
+    file transfer may invoke file-transfer plug-ins.  If it does, the hold
+    subcodes may additionally be 62 (ETIME), if the file-transfer plug-in
+    timed out; or the exit code of the plug-in shifted left by eight bits,
+    otherwise.
```

Comment:
```suggestion
.. note::
    For hold codes 12 [TransferOutputError] and 13 [TransferInputError]:
    file transfer may invoke file-transfer plug-ins.  If it does, the hold
    subcodes may additionally be 62 (ETIME), if the file-transfer plug-in
    timed out; or the exit code of the plug-in shifted left by eight bits,
    otherwise.
```